File: Data Structures & Algorithms/permutations/submission-1.java
class Solution {
    public List<List<Integer>> permute(int[] nums) {
        if(nums.length==0){
            return Arrays.asList(new ArrayList<>());
        }

        List<List<Integer>> perm = permute(
            Arrays.copyOfRange(nums,1,nums.length)
        );
        List<List<Integer>> res= new ArrayList<>();
        for(List<Integer> p : perm){
            for(int i =0 ; i<=p.size();i++){

                List<Integer> copy = new ArrayList<>(p);
                copy.add(i,nums[0]);
                res.add(copy);

            }


        }
        return res ;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n = nums.size();
        vector<int> res ; 
        for(int i = 0 ; i<n ; i++){
            int prod = 1 ;
            for(int j = 0 ; j<n ; j++){
                if(j!=i){
                prod*=nums[j];
            }
            }
            res.push_back(prod);   
        }
        return res ; 
    }
};
